app/domain/authentication/authn_k8s/validate_pod_request.rb

```diff
@@ -72,7 +72,11 @@ def host_can_access_service?
       end
 
       def host
-        @host ||= @resource_class[k8s_host.conjur_host_id]
+        unless defined? @host
+          @host = @resource_class[k8s_host.conjur_host_id]
+        end
```

Comment:
I believe you can replace this with the single line:
```ruby
@host ||= @resource_class[k8s_host.conjur_host_id]
```
Also even though it would be removed by the change suggested above, note that `defined? @host` should be `defined?(@host)` per [the ruby style guide's recommendation for method calls](https://github.com/rubocop-hq/ruby-style-guide#method-invocation-parens).  This should be getting caught by code climate but it may not be working.

Reply:
this is the whole fix. your suggestion was the original code but it's not good because `@resource_class[k8s_host.conjur_host_id]` can be nil. I want to raise an error in that case.